### PR TITLE
fix rule view

### PR DESCRIPTION
**Pull request by carl-zzz**
RuleDAO view has "Cannot read property 'find' of undefined" error. This will fix the error. 

```
Rule.js:215 Uncaught (in promise) TypeError: Cannot read property 'find' of undefined
    at Object.tableCellFormatter (Rule.js:215)
    at Object.format (FnFormatter.js:19)
    at Object.callOn (Element.js:1759)
    at Object.<anonymous> (UnstyledTableView.js:431)
    at Array.forEach (<anonymous>)
    at Object.forEach (Element.js:1782)
    at Object.<anonymous> (UnstyledTableView.js:427)
    at Object.addRow (Element.js:1713)
    at Element.js:591
```

### CODE REVIEW

**Comment by jlhughes** [comment]
Is this because Rule.js does not import userDAO?

**Comment by carl-zzz** [comment]
> Is this because Rule.js does not import userDAO?

@jlhughes Both work... Which way is better?

File: src/foam/nanos/ruler/Rule.js [hunk 1]
```diff
@@ -212,7 +212,7 @@
       name: 'createdBy',
       visibility: 'RO',
       tableCellFormatter: function(value, obj) {
-        obj.userDAO.find(value).then(function(user) {
+        this.__subSubContext__.userDAO.find(value).then(function(user) {
```
Comment (kgrgreer) [inline src/foam/nanos/ruler/Rule.js]: Best not to mess around with contexts directly. They're meant to be an internal thing and only used when you have no other option. Easier and better to just declare that you import userDAO.

**Comment by kgrgreer** [comment]
Please resolve merge conflict.

**Pull request merged**